### PR TITLE
Removed unused KMS permissions for nodePool role

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:
Removed KMS permissions not required to encrypt EC2 root volumes (ref: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSEncryption.html#ebs-encryption-permissions)

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by vincepri** [comment]
/retitle Removed unused KMS permissions for nodePool role

**Comment by enxebre** [comment]
I've seen inconsistent docs in aws e.g. https://docs.aws.amazon.com/autoscaling/ec2/userguide/key-policy-requirements-EBS-encryption.html
I also believe list/revoke Grant were needed in the past for the customer key scenario.

This is originally coming from https://bugzilla.redhat.com/show_bug.cgi?id=1815219

Let's make sure we update https://github.com/openshift/machine-api-operator/blob/d3973b45c2b6c477a803c9ed19cf8b8142e451a2/install/0000_30_machine-api-operator_00_credentials-request.yaml#L45-L62 consistently cc @JoelSpeed 

Hopefully e2e will catch if the change produce a regression

/approve

**Comment by lucasponce** [comment]
cc @oriAdler as well

File: api/v1beta1/hostedcluster_types.go [hunk 1]
```diff
@@ -1324,8 +1324,7 @@ type AWSRolesRef struct {
 	// 	  	"Effect": "Allow",
 	// 	  	"Action": [
 	// 	  		"kms:Decrypt",
-	// 	  		"kms:Encrypt",
-	// 	  		"kms:GenerateDataKey",
+	// 	  		"kms:ReEncrypt",
```
Comment (jharrington22) [inline api/v1beta1/hostedcluster_types.go:1327]: This should be `kms:ReEncrypt*` there is no ReEncrypt action see the AWS [reference](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awskeymanagementservice.html) for KMS actions.

And we should only require `GenerateDataKeyPairWithoutPlaintext`

**Review by jharrington22 — CHANGES_REQUESTED** [review]
We should also update the [KMS provider policy](https://github.com/openshift/hypershift/blob/main/api/v1alpha1/hostedcluster_types.go#L1784) to have `GenerateDataKeyPairWithoutPlaintext`

File: api/v1beta1/hostedcluster_types.go (hunk 1 above)
Comment (muraee) [reply api/v1beta1/hostedcluster_types.go:1327]: > We should also update the [KMS provider policy](https://github.com/openshift/hypershift/blob/main/api/v1alpha1/hostedcluster_types.go#L1784) to have `GenerateDataKeyPairWithoutPlaintext`

This role is used for etcd encryption, I think it needs a different set of perms cc @enxebre

Comment (muraee) [reply api/v1beta1/hostedcluster_types.go:1327]: I am following the docs [here ](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSEncryption.html#ebs-encryption-permissions) and it just mentions `kms:ReEncrypt`. Our e2e are also passing, so it does actually exist.
AWS docs seems to have some inconsistencies.

**Pull request merged**